tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -1645,6 +1663,35 @@ uint32_t SPIRVEmitter::processBinaryOp(const Expr *lhs, const Expr *rhs,
   return 0;
 }
 
+void SPIRVEmitter::initOnce(std::string varName, uint32_t varPtr,
+                            const Expr *varInit) {
+  const uint32_t boolType = theBuilder.getBoolType();
+  varName = "init.done." + varName;
+
+  // Create a file/module visible variable to hold the initialization state.
+  const uint32_t initDoneVar = theBuilder.addFileVar(
```

Comment:
I understand why you have done this... It's kind of ugly, but it works. And I can't think of any other way to do it....

* Is this safe from race conditions? (multiple workers writing to `init_done_c` at the same time?)
* How does glslang deal with this kind of initialization? Do they also use a boolean variable to make sure it is only initialized once?